File: Runs/Java_repo_codegemma:7b/Test_5.java
public class Test_5 {

    public static void main(String[] args) {
            unsigned int u_val = 4294967295L;
            System.out.printf("Original: %u\n", u_val);
            u_val = u_val + 1;
            System.out.printf("Overflow ( + 1): %u\n", u_val);
            u_val = u_val - 5;
            System.out.printf("Underflow ( - 5): %u\n", u_val);
        }

}

File: C_repo/test_5.c
#include <stdio.h>

int main() {
    unsigned int u_val = 4294967295; // (2^32 - 1)
    
    printf("Original: %u\n", u_val);
    
    // Test overflow
    u_val = u_val + 1;
    printf("Overflow ( + 1): %u\n", u_val); // Wraps to 0
    
    // Test underflow
    u_val = u_val - 5;
    printf("Underflow ( - 5): %u\n", u_val); // Wraps to 4294967291
    
    return 0;
}